libs/wallet/src/connectors/vega-connector.ts

```diff
@@ -1,68 +1,93 @@
 import { WalletClientError } from '@vegaprotocol/wallet-client';
-import type * as Schema from '@vegaprotocol/types';
-import type { PeggedReference } from '@vegaprotocol/types';
+import type { LiquidityProvisionSubmission as LiquidityProvisionBody } from '@vegaprotocol/protos/dist/vega/commands/v1/LiquidityProvisionSubmission';
+import type { DelegateSubmission } from '@vegaprotocol/protos/dist/vega/commands/v1/DelegateSubmission';
+import type { UndelegateSubmission } from '@vegaprotocol/protos/dist/vega/commands/v1/UndelegateSubmission';
+import type { OrderSubmission as OriginalOrderPosition } from '@vegaprotocol/protos/dist/vega/commands/v1/OrderSubmission';
+import type { OrderCancellation as OriginalOrderCancellation } from '@vegaprotocol/protos/dist/vega/commands/v1/OrderCancellation';
+import type { OrderAmendment as OriginalOrderAmendment } from '@vegaprotocol/protos/dist/vega/commands/v1/OrderAmendment';
+import type { VoteSubmission } from '@vegaprotocol/protos/dist/vega/commands/v1/VoteSubmission';
+import type { WithdrawSubmission as OriginalWithdrawSubmission } from '@vegaprotocol/protos/dist/vega/commands/v1/WithdrawSubmission';
+import type { ProposalSubmission as OriginalProposalSubmission } from '@vegaprotocol/protos/dist/vega/commands/v1/ProposalSubmission';
+import type { BatchMarketInstructions as OriginalBatchMarketInstructions } from '@vegaprotocol/protos/dist/vega/commands/v1/BatchMarketInstructions';
+import type { Transfer } from '@vegaprotocol/protos/dist/vega/commands/v1/Transfer';
+import type { ProposalTerms } from '@vegaprotocol/protos/dist/vega/ProposalTerms';
+import type { WithdrawExt } from '@vegaprotocol/protos/dist/vega/WithdrawExt';
+
+// generated type doesn't match in many places with current implementation. It has to be overwrite here
+// and re-exported for saving consistency.
+export type ProposalSubmission = Omit<
```

Comment:
I don't want to introduce these overriding types. The types exported here should be exactly as the wallet requires them. If there are differences we should make sure any transactions sent to the wallet conform to the interface

Reply:
I see now why we need these :). Can we simplify it though?

for example I think is more clear
```
export type OrderSubmission = SetOptional<
  protos.commands.v1.OrderSubmission.OrderSubmission,
  'reference' | 'peggedOrder' | 'price' | 'expiresAt'
>;
```

Reply:
done